pkg/config/config.go

```diff
@@ -433,6 +433,10 @@ type RuntimeConfig struct {
 	// Negative values indicate that the log file won't be truncated.
 	LogSizeMax int64 `toml:"log_size_max"`
 
+	// NoSyncLog if enabled will disable fsync on log rotation and container exit.
+	// This can improve performance but may result in date loss on hard system crashes.
+	NoSyncLog bool `toml:"no_sync_log"`
```

Comment:
@cri-o/cri-o-maintainers I wonder if this should be a member of the runtime config structure, as it doesn't work for runtime pod or vm right now

Reply:
I just checked other log related settings in this struct and they seem to have mixed scopes as well
- `LogSizeMax` seems to apply to oci and pod
- `LogToJournald` seems to apply to oci only

Would it be acceptable to document this correctly?
E.g. appending `Applies to oci containers only` to all doc strings for the setting?

Reply:
We have, indeed, overloaded **RuntimeConfig** a little bit. I still am of a belief that **LogLevel** should not be there, for example. I suppose it was also done with some of the options for convenience since it's easier to access some of these depending on which objects are being passed around...

@haircommander, I think it would be fine... for now. We then can figure out how to clean this up.

Thoughts?

Reply:
sorry by runtime config I mean `crio.runtime.runtimes` runtime (talk about overloaded!) . this only works for runtime type OCI, and I'd love to see it as a member there instead

Reply:
Ah, ok, so
* Move it to RuntimeHandler
* Add validation that it only applies to OCI

The thing I haven't figured out is what to do about the cli option in that case.
But I'll update this PR once I get that sorted.

Reply:
I personaly don't think the cli option is needed. Do you really want it @rtreffer ?

Reply:
Well, it was _convinient_, but looks like it might be annoying to expose it as a flag, so I am in favor of dropping it for simplicity.